subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -179,26 +179,29 @@ private String getFilename(Path dir) {
         public FileVisitResult visitFile(Path file, @Nullable BasicFileAttributes attrs) {
             String name = internedToString(file.getFileName());
             if (isAllowed(file, name, false, attrs, builder.getRelativePath())) {
-                if (attrs == null) {
-                    throw new FileSnapshottingException(String.format("Cannot read file '%s': not authorized.", file));
-                }
```

Comment:
I don't think we should remove this. This is about not being able to read the parent directory (where the attributes of the children are stored), which we shouldn't handle the same way as not being able to read the child itself.

Reply:
I'll investigate, thanks

Reply:
Done, thanks for noticing

Reply:
cc: @wolfs